ETL/Pipeline/extract.py

```diff
@@ -0,0 +1,98 @@
+"""Extract Script: Pulls current price and sale status from ASOS API"""
+from datetime import datetime
+import logging
+
+import requests
+from lambda_multiprocessing import Pool
+
+
+def configure_log() -> None:
+    """Configures log output"""
+    logging.basicConfig(
+        level=logging.INFO,
+        format='%(asctime)s - %(levelname)s - %(message)s'
+    )
+
+
+def get_url(product_id: int) -> str | None:
+    """Returns the API URL for a given product on the ASOS website."""
+    if not isinstance(product_id, int):
+        logging.error('Product ID must be a integer to get the url.')
+        return None
+
+    return f"https://www.asos.com/api/product/catalogue/v4/stockprice?productIds=\
+        {product_id}&store=COM&currency=GBP&keyStoreDataversion=ornjx7v-36&country=GB"
+
+
+def get_product_info(product_data: int, headers: dict) -> dict | None:
+    """Gets the price information for a specified product from the ASOS API."""
+    price_endpoint = get_url(product_data['product_id'])
+
+    if not price_endpoint:
+        return None
+
+    response = requests.get(price_endpoint, headers, timeout=60).json()
+
+    if 'errorCode' in response or not response:
+        logging.error('No valid ProductIds requested')
+        return None
+
+    return response[0]
+
+
+def get_current_price(product_info: dict) -> int | None:
+    """Extracts the current price of the product from the product information."""
+    try:
+        return product_info["productPrice"]["current"]["value"]
+    except KeyError as e:
+        logging.error(f"Error getting current price:")
```

Comment:
```suggestion
        logging.error("Error getting current price: %s", e)
```

Reply:
Could you explain this suggestion? Why is it better?

Reply:
You haven't included the error message and this is how you do it for logging